runtime/src/snapshot_utils.rs

```diff
@@ -750,14 +1143,36 @@ fn parse_snapshot_archive_filename(archive_filename: &str) -> Option<(Slot, Hash
     })
 }
 
+/// Parse an incremental snapshot archive filename into its base Slot, actual Slot, Hash, and Archive Format
+fn parse_incremental_snapshot_archive_filename(
+    archive_filename: &str,
+) -> Option<(Slot, Slot, Hash, ArchiveFormat)> {
+    let regex = Regex::new(INCREMENTAL_SNAPSHOT_ARCHIVE_FILENAME_REGEX);
+
+    regex.ok()?.captures(archive_filename).and_then(|captures| {
+        let base_slot = captures.get(1).map(|x| x.as_str().parse::<Slot>())?.ok()?;
```

Comment:
maybe, there is no `captures.iter()` or something similar? I like multiple `capture_iter.next()`s than `.get(1)`/`.get(2)`/`.get(3)`.... (no magic numbers if possible..)

Reply:
I agree; no magic numbers if possible! I'll take a look to see if there's another way (i.e. `.iter()`) to accomplish this.

Reply:
Done!